osquery/dispatcher/scheduler.cpp

```diff
@@ -146,12 +147,11 @@ inline void launchQuery(const std::string& name, const ScheduledQuery& query) {
     diff_results.added = std::move(sql.rows());
   }
 
-  if (query.options.count("removed") && !query.options.at("removed")) {
+  if (query.shouldClearRemovedRows()) {
     diff_results.removed.clear();
   }
 
-  if (diff_results.added.empty() && diff_results.removed.empty()) {
-    // No diff results or events to emit.
+  if (diff_results.hasResults()) {
```

Comment:
`hasResults` name looks a bit confusing here, I think. Probably you mean `hasNoResults`, right?